### PR TITLE
documenting functions with multiple clauses.

### PR DESCRIPTION
**Pull request by stephenmoloney**
#### Description

Minor addition to the writing-documentation.md file to explain best practice for documenting functions with multiple clauses.

#### Motivation

I often found myself in a dilemma on how to document multiple clauses. See [gist](https://gist.github.com/stephenmoloney/16efb1e7168fdbeb74f9) as an example.

#### Solution 

Add to elixir docs best practice guideline.

## Important !!

Please check that this change to the docs actually reflects best practice before merging and that this change is not misleading! 

Thanks ! :+1:

### CODE REVIEW

File: lib/elixir/pages/Writing Documentation.md [hunk 1]
```diff
@@ -48,6 +48,38 @@ The compiler will infer this argument as `map`. Sometimes the inference will be
 
     def size(map)
 
+## Functions with multiple clauses
+
+When documenting a function with multiple clauses, it is recommended to add the documentation
+and typespecs to the function head rather than adding @doc and @spec to each clause of
+the function.
```
Comment (whatyouhide) [inline lib/elixir/pages/Writing Documentation.md]: Could you not wrap lines and place this sentence over a single line? Just to be coherent with the style of the rest of this document :smiley:

Comment (whatyouhide) [inline lib/elixir/pages/Writing Documentation.md]: Also, I would rephrase this as:

> When documenting a function with multiple clauses, it is recommended to add a bodyless function head and add `@doc`s and `@spec`s to it.

This is also necessary when the function takes one or more optional arguments because you wouldn't know which clause to add the `\\` to :).

File: lib/elixir/pages/Writing Documentation.md [hunk 2]
```diff
@@ -48,6 +48,38 @@ The compiler will infer this argument as `map`. Sometimes the inference will be
...
+      Float.round(total, 2)
+    end
+
+    def add_tax(amount, country: country) do
+      total = amount + calc_tax(amount, get_tax_rate(country))
+      Float.round(total, 2)
+    end
+
+    defp calc_tax(amount, rate), do: amount * rate
+    defp get_tax_rate("Ireland"), do: 0.23
```
Comment (whatyouhide) [inline lib/elixir/pages/Writing Documentation.md]: Do you think we could reduce the complexity of this function? I'm afraid it overshadows what we're trying to showcase here :). We could use whatever simple function that requires a pattern match.

**Comment by whatyouhide** [comment]
Great job @stephenmoloney :clap: I added a couple of comments, let me know what you think!

**Comment by josevalim** [comment]
Great, thank you! What if we just add a new bullet under the "Recommendations" section? I don't think we need to expand with examples. :)

**Comment by stephenmoloney** [comment]
@whatyouhide You are right about that. Thanks for suggestions.

**Comment by stephenmoloney** [comment]
@whatyouhide @josevalim 
I added both suggestions and obviously both should not be put in.
In seeing both side by side, which one would you go for? 
1. The full paragraph and example.
or
2. Just the bullet point.

**Comment by stephenmoloney** [comment]
I reduced it to just the bullet point. It should suffice. 
Thanks for clarifications & suggestions.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by stephenmoloney** [comment]
:tada:
